### ISSUE TITLE
Event delivery is wrong sometimes

### ISSUE THREAD

**Issue by lppedd** [quarkiverse/quarkus-github-app#460]
Currently I have two event listeners for PRs:

```kotlin
@EventHandler
fun onPullRequestEvent(@Opened @Reopened @Synchronize payload: GHEventPayload.PullRequest) {
  val pullRequest = payload.pullRequest

  if (pullRequest.isOpen) {
    checkPullRequest(pullRequest)
  }
}

@EventHandler
fun onPullRequestLabelEvent(@Labeled @Unlabeled payload: GHEventPayload.PullRequest) {
  val pullRequest = payload.pullRequest

  if (pullRequest.isOpen) {
    checkPullRequest(pullRequest)
  }
}
```

smee.io shows the delivered events when I open a new PR via a GitHub App, assigning labels just after opening.  
You can see "opened" comes before "labeled".

![image](https://user-images.githubusercontent.com/19871649/229515229-a61f7699-5dde-419d-9deb-5556348aa889.png)

However, debugging I've found out that the "opened" event is received later in time, messing with check runs.

```
Label: executor-thread-2, 1680526155762 (System.currentTimeMillis)
Label: executor-thread-3, 1680526156010 (System.currentTimeMillis)
Opened: executor-thread-1, 1680526157728 (System.currentTimeMillis)
```

**Comment by gsmet** [comment]
Unfortunately, I don't think we have the possibility to guarantee a strong ordering. We receive the events on various threads and they are treated in parallel.

@yrodiere WDYT?

**Comment by lppedd** [comment]
@gsmet couldn't they be parallelized based on the object id? Just guessing as I don't know enough about how you do it.

If this cannot be done, it's probably better to make it clear somewhere, or offer a flag to entirely switch off parallelization.

**Comment by yrodiere** [comment]
> @yrodiere WDYT?

We could probably guarantee that event processing *starts* in the order we receive events, but it would still be possible for the first event to be processed slower than the second one. If we add the requirement that events for the same repository are processed one at a time (put it through a queue or an equivalent reactive solution), it could solve the problem... except that we could still possibly receive events in the wrong order due to varying network delays.

Really, the solution would be for your app to be resilient to events coming in the wrong order, because there'll always be a chance that it happens. How to be resilient would depend on the exact stuff you're doing, though, and I don't know what that is.

**Comment by lppedd** [comment]
@yrodiere that is a fair point.

The "starts in order" approach might be a good trade off. At that point synchronization is up to me.

I don't care about the "finished processing" part, as it probably adds complexity for no big benefit.

And regarding how I solved this problem: most objects are "refreshable", so before checking requirements I simply ask the GitHub API for updated data.

**Comment by lppedd** [comment]
@gsmet sounds good. Worth adding this note in the docs?
